exercises/exercise-11/declarations/stats/index.d.ts

```diff
@@ -1,3 +1,16 @@
 declare module 'stats' {
-    export function getMaxIndex(input: unknown, comparator: unknown): unknown;
+    type ComparatorFn<T, R> = (input: T[], comparator: (a: T, b: T) => number) => R;
+
+    type CompArgs<T> = [T[], (a: T, b: T) => number];
+
+    function GetIndexFn<T>(input: T[], comparator: (a: T, b: T) => number): number;
+    function GetElementFn<T>(input: T[], comparator: (a: T, b: T) => number): T | null;
+
+    export const getMaxIndex: typeof GetIndexFn;
+    export const getMinIndex: typeof GetIndexFn;
+    export const getMedianIndex: typeof GetIndexFn;
+    export const getMaxElement: typeof GetElementFn;
+    export const getMinElement: typeof GetElementFn;
+    export const getMedianElement: typeof GetElementFn;
+    export const getAverageValue: typeof GetElementFn;
```

Comment:
I think `getAverageValue` should be different from others
```
export function getAverageValue<T>(input: T[], getValue: (a: T) => number): number | null;
```

Reply:
I am agree，there different comparator in **getAverageVal**

Reply:
i think getAverageValue shouldn't be null
```js
 export function getAverageValue<T>(
    input: T[],
    getValue: (item: T) => any
  ): number;
}
```